AwsCryptographicMaterialProviders/runtimes/java/src/testExamples/java/software/amazon/cryptography/example/hierarchy/concurrent/ConcurrentConditionCheckWriteTest.java

```diff
@@ -0,0 +1,179 @@
+package software.amazon.cryptography.example.hierarchy.concurrent;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.GetItemResponse;
+import software.amazon.awssdk.services.dynamodb.model.TransactWriteItem;
+import software.amazon.awssdk.services.dynamodb.model.TransactWriteItemsRequest;
+import software.amazon.awssdk.services.dynamodb.model.TransactWriteItemsResponse;
+import software.amazon.awssdk.services.dynamodb.model.TransactionCanceledException;
+import software.amazon.awssdk.utils.ImmutableMap;
+import software.amazon.cryptography.example.Constants;
+import software.amazon.cryptography.example.DdbHelper;
+import software.amazon.cryptography.example.Fixtures;
+
+// These concurrent tests check that the
+public class ConcurrentConditionCheckWriteTest {
+
+  private static final Integer threadCount = 5;
+  private static final String mLockedId = "concurrency-test-write-key";
+  private static final Map<String, String> INDEX_EXPR_ATT_NAMES =
+    ImmutableMap.of("#pk", "branch-key-id");
+
+  private static final List<String> identifiers = Collections.unmodifiableList(
+    Arrays.asList("1", "2", "3", "4", "5")
+  );
+  private Map<String, DynamoDbClient> ddbClientToThreadId;
+  private static Map<String, String> indexToThreadId, threadIndexToTimestamp;
+  private ConcurrentLinkedDeque<String> unpickedIndices;
+
+  @BeforeClass
+  public void setup() {
+    ddbClientToThreadId = new ConcurrentHashMap<>(6, 1, threadCount);
+    identifiers.forEach(id ->
+      ddbClientToThreadId.put(id, DynamoDbClient.create())
+    );
+    indexToThreadId = new ConcurrentHashMap<>(6, 1, threadCount);
+    unpickedIndices = new ConcurrentLinkedDeque<>(identifiers);
+    threadIndexToTimestamp = new ConcurrentHashMap<>(6, 1, threadCount);
+  }
+
+  @AfterClass
+  public void teardown() {
+    identifiers.forEach(id ->
+      DdbHelper.deleteKeyStoreDdbItem(
+        mLockedId,
+        "branch:ACTIVE",
+        Fixtures.TEST_KEYSTORE_NAME,
+        DynamoDbClient.create(),
```

Comment:
No. Do not create a DDB Client in a for loop.
Create the client on line 58, and pass it in.